pkg/engine/parser/parser.go

```diff
@@ -638,16 +639,48 @@ func scriptJSFileRegexParser(resp *navigation.Response) (navigationRequests []*n
 	if !(strings.HasSuffix(resp.Resp.Request.URL.Path, ".js") || strings.HasSuffix(resp.Resp.Request.URL.Path, ".css") || strings.Contains(contentType, "/javascript")) {
 		return
 	}
-
 	endpoints := utils.ExtractRelativeEndpoints(string(resp.Body))
 	for _, item := range endpoints {
 		navigationRequests = append(navigationRequests, navigation.NewNavigationRequestURLFromResponse(item, resp.Resp.Request.URL.String(), "js", "regex", resp))
 	}
+
+	if strings.Contains(string(resp.Body), "//# sourceMappingURL=") {
+		url := resp.Resp.Request.URL
+		url.RawQuery = ""
+		navigationRequests = append(navigationRequests, navigation.NewNavigationRequestURLFromResponse(url.String()+".map", resp.Resp.Request.URL.String(), "js", "eval", resp))
+	}
```

Comment:
@ShubhamRasal , not sure if there is a need to remove any existing parameters . we can clone any url struct using 
```go
	temp := urlutil.URL{URL: resp.Resp.Request.URL}
	cloned := temp.Clone()
```
and we can add to cloned.Path directly